src/windows/osd.c

```diff
@@ -402,65 +402,104 @@ ssize_t sendmsg(SOCKET sd, struct msghdr *msg, int flags)
 	goto fn_complete;
 }
 
+void sort_list_of_adapters(PIP_ADAPTER_ADDRESSES head)
+{
+	PIP_ADAPTER_ADDRESSES ptr = head, tmp = NULL, prev = NULL;
+	bool sorted;
+
+	if (!head)
+		return;
+
+	do {
+		sorted = false;
+		ptr = head;
+		for (ptr = head; ptr->Next; ptr = ptr->Next) {
+			if (ptr->Ipv4Metric < ptr->Next->Ipv4Metric) {
+				tmp = ptr;
+				ptr = tmp->Next;
+				tmp->Next = ptr->Next;
+				ptr->Next = tmp;
+				if (tmp == head)
+					head = ptr;
+				else
+					prev->Next = ptr;
+				sorted = true;
+			}
+			ptr = ptr->Next;
+		}
+	} while (sorted);	
+}
```

Comment:
I think the easiest solution here may be to do something like this:
```
sort_adapters(&head) /* pass in address of head, since we need to change it */
{
    PIP_ADAPTER *new_head = NULL, *item;
    while (*head) {
        item = *head;
        *head = (*head)->Next;
        insert_before_first_match(&new_head, item);
    }
    *head = new_head;
}
````
For insert_before_first_match, see ofi_list.h :: slist_insert_before_first_match() for an example.  You won't need a comparison function.  Just do the IPv4Metric check.